### PR TITLE
Feature/ab2d 2109 separation for eob data serialization

### PR DESCRIPTION
**Pull request by lailasharshar**
**JIRA Tickets:**

[AB2D-2109](https://jira.cms.gov/browse/AB2D-2109) - Moved serialization from the EOB search Future to the contract processor.

### What Does This PR Do?

 First step in separation of the different components from worker - job management, EOB searching and data serialization. Instead of returning Void in the Future, it now returns a response object with the EOB values. Validation of the patients against the contract was also moved to the contract processor.

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [X] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [X] Code checked for PHI/PII exposure

### CODE REVIEW

File: worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java [hunk 1]
```diff
@@ -87,115 +80,67 @@
...
-    }
-
-    private void close(StreamHelper helper) {
-        if (helper != null) {
-            try {
-                helper.close();
-            } catch (Exception ex) {
-                log.error("Unable to close the helper", ex);
+            while (!futureHandles.isEmpty()) {
+                sleep();
```
Comment (nflearl) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java:116]: Having to physically sleep the thread while awaiting future completion is a red flag whereas a more reactive model would simply have a method/function invoked where complete.

I am not advocating doing anything additionally ATM, but have added it as a consideration for https://jira.cms.gov/browse/AB2D-2157.

File: worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java [hunk 2]
```diff
@@ -87,115 +80,67 @@
...
+                numberOfEobs += processHandles(futureHandles, progressTracker, patients, helper);
+                if (hasJobBeenCancelled(jobUuid)) {
```
Comment (nflearl) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java:118]: Another artifact of the current architecture, having to poll the job table to see if the API application has performed a cancellation.

File: worker/src/test/java/gov/cms/ab2d/worker/processor/ContractProcessorInvalidPatientTest.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    @Test
+    void testWriteNullErrors() throws IOException {
+        Job job = new Job();
+        job.setJobUuid(jobId);
+        StreamHelper helper = new TextStreamHelperImpl(
+                tmpDirFolder.toPath(), contractId, 2000, 10, eventLogger, job);
+        ((ContractProcessorImpl) cut).writeExceptionToContractErrorFile(
+                helper, null, new RuntimeException("Exception"));
+        assertThrows(NoSuchFileException.class, () -> Files.readString(Path.of(tmpDirFolder.getAbsolutePath() + "/" + contractId + "_error.ndjson")));
```
Comment (nflearl) [inline worker/src/test/java/gov/cms/ab2d/worker/processor/ContractProcessorInvalidPatientTest.java:140]: Use  File.separator or some other environmentally defined path seperation.  Forward slash does not compare will on my box.

File: worker/src/test/java/gov/cms/ab2d/worker/processor/ContractProcessorInvalidPatientTest.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    @Test
+    void testWriteErrors() throws IOException {
+        Job job = new Job();
+        job.setJobUuid(jobId);
+        String val = "Hello World";
+        StreamHelper helper = new TextStreamHelperImpl(
+                tmpDirFolder.toPath(), contractId, 2000, 10, eventLogger, job);
+        ((ContractProcessorImpl) cut).writeExceptionToContractErrorFile(
+                helper, val, new RuntimeException("Exception"));
+        String result = Files.readString(Path.of(tmpDirFolder.getAbsolutePath() + "/" + contractId + "_error.ndjson"));
```
Comment (nflearl) [inline worker/src/test/java/gov/cms/ab2d/worker/processor/ContractProcessorInvalidPatientTest.java:128]: Ditto from below.

**Review by nflearl — CHANGES_REQUESTED** [review]
Only requesting fixes in test to be platform agnostic when creating files.

File: worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java (hunk 1 above)
Comment (lailasharshar) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java:116]: We definitely need a rework of the whole contract job management

**Review by wnyffenegger — APPROVED** [review]

**Review by nflearl — APPROVED** [review]

**Review by adaykin — APPROVED** [review]

**Pull request merged**